public/store/dataset/mutations.ts

```diff
@@ -334,12 +335,21 @@ export const mutations = {
 
   // sets the current selected data into the store
   setIncludedTableData(state: DatasetState, tableData: TableData) {
-    state.includedSet.tableData = tableData;
+    state.includedSet.tableData = Object.freeze(tableData);
```

Comment:
Same comment for `Object.seal()`, a small comment might avoid the future issue by someone trying to be smart and removing it.